api/permissions.py

```diff
@@ -27,3 +29,15 @@ def has_permission(self, request, view):
         assert False, ('TokenHasScope requires either the'
                        '`oauth2_provider.rest_framework.OAuth2Authentication` authentication '
                        'class to be used.')
+
+
+class IsDeletedPremissions(BasePermission):
+    """
+    Permission check for deleted objects.
+    """
+    def has_object_permission(self, request, view, obj):
+        TargetObject = AbstractCreativeWork.objects.filter(id=obj.id).first()
+        if hasattr(TargetObject, 'is_deleted'):
```

Comment:
As there are no other code paths, this would be better expressed using `and`. 
if hasattr(...) and ...is_deleted:`